packages/docusaurus/lib/server/load/routes.js

```diff
@@ -5,98 +5,159 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const {generateChunkName} = require('@docusaurus/utils');
+const {genChunkName, docuHash} = require('@docusaurus/utils');
 const {stringify} = require('querystring');
 
 async function loadRoutes(pluginsRouteConfigs) {
-  const imports = [
+  const routesImports = [
     `import React from 'react';`,
     `import Loadable from 'react-loadable';`,
     `import Loading from '@theme/Loading';`,
     `import NotFound from '@theme/NotFound';`,
   ];
-
+  // Routes paths. Example: ['/', '/docs', '/blog/2017/09/03/test']
   const routesPaths = [];
-  const addRoutesPath = permalink => {
-    if (permalink && !/:|\*/.test(permalink)) {
-      routesPaths.push(permalink);
+  const addRoutesPath = routePath => {
+    routesPaths.push(routePath);
+  };
+  // Mapping of routePath -> metadataPath. Example: '/blog' -> '@generated/metadata/blog-c06.json'
+  const routesMetadataPath = {};
+  const addRoutesMetadataPath = routePath => {
+    const fileName = `${docuHash(routePath)}.json`;
+    routesMetadataPath[routePath] = `@generated/metadata/${fileName}`;
```

Comment:
Can we generate the metadata in the plugins dir instead? It doesn't seem like a scalable design to put all metadata into one single metadata folder. The concerns feel like they're cross-cutting again and it feels like we're blurring the lines between plugins and the monolithic v2.

Reply:
Actually the metadata is related to route level.

At this routes.js file we also dont know which route come from which plugin because our API is only

‘addRoute(path, component, metadata)’

Need to figure out the correct plugin API before splitting it up.